### PR TITLE
LPD-33373 fix tests

### PR DESCRIPTION
**Pull request by thiago-buarqque**
Ticket: https://liferay.atlassian.net/browse/LPD-33373

These tests started failing because now portal starts up with language overrides defined by the Objects functionality. The keys are similar to the ones in the image
image

# Tests migration

- `LanguageOverrideExportImport#CanExportMultipleLanguageKeysWithOverride` -> Covered by "LPD-33373 assert that overriden translations can be exported"
- `LanguageOverride#CanFilterOverridesByAnyLanguage` -> Covered by "LPD-33373 assert that overriden translations can be filtered"
-  `LanguageOverride#NoOverrideRelatedInfosDisplayForOOTBLanguageKey` -> Covered by "LPD-33373 assert that default and overriden translations show up when no filters are applied"
- `LanguageOverride#CanFilterLanguageKeysByLocaleViaLanguageSelector` -> Covered by "LPD-33373 assert that overriden translations can be filtered"

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:playwright

**Comment by thiago-buarqque** [comment]
Unrelated CI errors, the tests added in this PR are passing

File: modules/test/playwright/pages/portal-language-override-web/LanguageOverridePage.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import {clickAndExpectToBeVisible} from '../../utils/clickAndExpectToBeVisible';
+import fillAndClickOutside from '../../utils/fillAndClickOutside';
+import {PORTLET_URLS} from '../../utils/portletUrls';
+import {waitForSuccessAlert} from '../../utils/waitForSuccessAlert';
+
+export type TLanguageOverride = {
+	key: string;
+	languageIds: string[];
+	values: string[];
```
Comment (ethib137) [inline modules/test/playwright/pages/portal-language-override-web/LanguageOverridePage.ts:16]: Two arrays that relate by index for languageIds and values seems less than intuitive. How about just having an object for values there the languageId is the key and the translation is the value.
```
{
  key: 'my-language-key',
  values: {
    ['en-us']: 'My Language key'
  }
}
```

File: modules/test/playwright/utils/fillAndClickOutside.ts [hunk 2]
```diff
@@ -10,9 +10,13 @@ import getRandomString from './getRandomString';
 export default async function fillAndClickOutside(
 	page: Page,
 	element: Locator,
-	content?: string
+	content?: string,
+	clickOutside: boolean = true
```
Comment (ethib137) [inline modules/test/playwright/utils/fillAndClickOutside.ts:14]: should be sorted

**Review by ethib137 — CHANGES_REQUESTED** [review]
Just a couple thoughts for changes. Nice work Thiago!

File: modules/test/playwright/tests/portal-language-override-web/languageOverride.spec.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {languageOverridePageTest} from '../../fixtures/LanguageOverridePageTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {TLanguageOverride as TTranslation} from '../../pages/portal-language-override-web/LanguageOverridePage';
```
[the remark this answers is not in the document]
Comment (thiago-buarqque) [reply modules/test/playwright/tests/portal-language-override-web/languageOverride.spec.ts:10]: My bad, I renamed it from TLanguageOverride to TTranslation using the VSCode rename tool and it did that.

File: modules/test/playwright/pages/portal-language-override-web/LanguageOverridePage.ts (hunk 1 above)
Comment (thiago-buarqque) [reply modules/test/playwright/pages/portal-language-override-web/LanguageOverridePage.ts:16]: Way clever, thanks

File: modules/test/playwright/tests/portal-language-override-web/languageOverride.spec.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+				},
+				{
+					languageId: 'es-ES',
+					value: getRandomString(),
+				},
+				{
+					languageId: 'pt-BR',
+					value: getRandomString(),
+				},
+			],
```
Comment (ethib137) [inline modules/test/playwright/tests/portal-language-override-web/languageOverride.spec.ts:36]: @thiago-buarqque , this is different than what I was suggesting, but also works. I was suggesting this:
```
values: {
	['en-US']: getRandomString(),
	['es-ES']: getRandomString(),
	['pt-BR']: getRandomString(),
}
```

My way is a little less verbose but yours is more explicit. They both have pros and cons so feel free to choose what you think is best in this case.

**Review by ethib137 — APPROVED** [review]
Looks good. I added one comment that you can choose whether or not you want to change. Thanks!

File: modules/test/playwright/tests/portal-language-override-web/languageOverride.spec.ts (hunk 4 above)
Comment (thiago-buarqque) [reply modules/test/playwright/tests/portal-language-override-web/languageOverride.spec.ts:36]: @ethib137 I prefer this way :sweat_smile: For me it's easier to understand. But thanks though

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:playwright

**Comment by thiago-buarqque** [comment]
ci:test:playwright

File: modules/test/playwright/utils/fillAndClickOutside.ts (hunk 2 above)
Comment (ethib137) [reply modules/test/playwright/utils/fillAndClickOutside.ts:14]: Yep, no need actually. I misunderstood the context when I added that comment. For JS we don't always require parameters of a function to be sorted. Thanks.

**Comment by ethib137** [comment]
ci:report:59188781

**Comment by ethib137** [comment]
Thanks @thiago-buarqque !

**Comment by ethib137** [comment]
ci:forward

**Pull request closed without merging**